File: Backend/internal/model/circle.go
package model

import "time"

type Circle struct {
	ID        string    `db:"id" json:"id"`
	Name      string    `db:"name" json:"name"`
	Emoji     string    `db:"emoji" json:"emoji"`
	Color     string    `db:"color" json:"color"`
	OwnerID   string    `db:"owner_id" json:"ownerId"`
	CreatedAt time.Time `db:"created_at" json:"createdAt"`
	UpdatedAt time.Time `db:"updated_at" json:"updatedAt"`
}

type CircleMember struct {
	ID       string    `db:"id" json:"id"`
	CircleID string    `db:"circle_id" json:"circleId"`
	UserID   string    `db:"user_id" json:"userId"`
	JoinedAt time.Time `db:"joined_at" json:"joinedAt"`
}

type CircleWithMembers struct {
	Circle
	Members []UserProfile `json:"members"`
}

type CircleDetail struct {
	ID          string        `json:"id"`
	Name        string        `json:"name"`
	Emoji       string        `json:"emoji"`
	Color       string        `json:"color"`
	OwnerID     string        `json:"ownerId"`
	MemberCount int           `json:"memberCount"`
	Members     []UserProfile `json:"members,omitempty"`
	CreatedAt   time.Time     `json:"createdAt"`
}

File: Android/core/core-domain/src/main/java/com/youkong/core/domain/model/Circle.kt
package com.youkong.core.domain.model

import kotlinx.datetime.Instant

data class Circle(
    val id: String,
    val name: String,
    val emoji: String,
    val color: String,
    val ownerId: String,
    val createdAt: Instant,
)

data class CircleDetail(
    val id: String,
    val name: String,
    val emoji: String,
    val color: String,
    val ownerId: String,
    val memberCount: Int,
    val members: List<UserProfile>? = null,
    val createdAt: Instant,
)

// 预设圈子颜色
object CircleColorPresets {
    const val PINK = "#EC4899"
    const val ORANGE = "#F97316"
    const val BLUE = "#3B82F6"
    const val GREEN = "#22C55E"
    const val PURPLE = "#8B5CF6"
    const val YELLOW = "#EAB308"

    val all = listOf(PINK, ORANGE, BLUE, GREEN, PURPLE, YELLOW)
}
